test/e2e/prometheus_test.go

```diff
@@ -5495,6 +5495,280 @@ func testPrometheusReconciliationOnSecretChanges(t *testing.T) {
+}
+
+func testPrometheusUTF8LabelSupport(t *testing.T) {
+	skipPrometheusTests(t)
+	t.Parallel()
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	_, err := framework.CreateOrUpdatePrometheusOperator(
+		context.Background(),
+		ns,
+		[]string{ns},
+		nil,
+		[]string{ns},
+		nil,
+		false,
+		true,
+		true,
+	)
+	require.NoError(t, err)
+
+	t.Run("UTF8RelabelConfigWithPrometheus3", func(t *testing.T) {
+		testUTF8RelabelConfigWithPrometheus3(t, ns)
+	})
+
+	t.Run("UTF8RelabelConfigWithPrometheus2", func(t *testing.T) {
+		testUTF8RelabelConfigWithPrometheus2(t, ns)
+	})
+
+	t.Run("UTF8ServiceMonitorLabelsWithPrometheus3", func(t *testing.T) {
+		testUTF8ServiceMonitorLabelsWithPrometheus3(t, ns)
+	})
+}
+
+func testUTF8RelabelConfigWithPrometheus3(t *testing.T, ns string) {
+	name := "prometheus-utf8-relabel"
+
+	prom := framework.MakeBasicPrometheus(ns, name, "test-app", 1)
+	prom.Spec.Version = operator.DefaultPrometheusVersion
+
+	_, err := framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, prom)
+	require.NoError(t, err)
+
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-service",
+			Namespace: ns,
+			Labels:    map[string]string{"group": "test-app"},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name: "metrics",
+					Port: 8080,
+				},
+			},
+			Selector: map[string]string{"app": "test"},
+		},
+	}
+	_, err = framework.KubeClient.CoreV1().Services(ns).Create(context.Background(), service, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	sm := &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "utf8-relabel-test",
+			Namespace: ns,
+			Labels:    map[string]string{"group": "test-app"},
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"group": "test-app"},
+			},
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					Port:     "metrics",
+					Interval: "30s",
+					RelabelConfigs: []monitoringv1.RelabelConfig{
+						{
+							SourceLabels: []monitoringv1.LabelName{"__name__"},
+							TargetLabel:  "服务",
+							Replacement:  ptr.To("测试服务"),
+							Action:       "replace",
+						},
+					},
+					MetricRelabelConfigs: []monitoringv1.RelabelConfig{
+						{
+							SourceLabels: []monitoringv1.LabelName{"job"},
+							TargetLabel:  "环境",
+							Replacement:  ptr.To("生产环境"),
+							Action:       "replace",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err = framework.MonClientV1.ServiceMonitors(ns).Create(context.Background(), sm, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// Verify that UTF-8 labels appear in the config with Prometheus 3.0+
+	err = wait.PollUntilContextTimeout(context.Background(), time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
```

Comment:
rather than checking the config can we configure a service monitor that scrapes a real target (e.g. Prometheus), query the metric in Prometheus and check that we get the expected labels?